src/backend/executor/bridge.ts

```diff
@@ -141,69 +159,66 @@ export class ExecutorBridge implements Disposable {
         }
 
         const filePaths = files.length
-            ? `--file ${files.map((uri) => `"${uri.fsPath}"`).join(' ')}`
-            : '';
+            ? ['--file', ...files.map((uri) => uri.fsPath)]
+            : [];
 
-        return [
-            `${ccPath} analyze`,
-            `"${ccCompileCmd}"`,
-            `--output "${reportsFolder}"`,
-            `${ccThreads ? '-j ' + ccThreads : ''}`,
-            `${ccArguments}`,
-            `${filePaths}`,
-        ].join(' ');
+        const args = [
+            'analyze', ccCompileCmd,
+            '--output', reportsFolder,
+        ];
+
+        if (ccThreads) {
+            args.push('-j', ccThreads);
+        }
+
+        args.push(...ccArguments, ...filePaths);
+
+        return args;
     }
 
-    public getLogCmdLine(): string | undefined {
+    public getLogCmdArgs(): string[] | undefined {
         if (!workspace.workspaceFolders?.length) {
             return undefined;
         }
 
         const workspaceFolder = workspace.workspaceFolders[0].uri.fsPath;
-
-        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
-            ?? 'CodeChecker';
         const ccFolder = getConfigAndReplaceVariables('codechecker.backend', 'outputFolder')
             ?? path.join(workspaceFolder, '.codechecker');
 
         // Use a predefined path here
         const ccCompileCmd = path.join(ccFolder, 'compile_commands.json');
 
         return [
-            `${ccPath} log`,
-            `--output "${ccCompileCmd}"`,
-            '--build "make"'
-        ].join(' ');
+            'log',
+            '--output', ccCompileCmd,
+            '--build', 'make'
+        ];
     }
 
-    public getParseCmdLine(...files: Uri[]): string | undefined {
+    public getParseCmdArgs(...files: Uri[]): string[] | undefined {
         if (!workspace.workspaceFolders?.length) {
             return undefined;
         }
 
-        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath') ?? 'CodeChecker';
         const reportsFolder = this.getReportsFolder();
 
         const filePaths = files.length
-            ? `--file ${files.map((uri) => `"${uri.fsPath}"`).join(' ')}`
-            : '';
+            ? ['--file', ...files.map((uri) => uri.fsPath)]
+            : [];
 
         return [
-            `${ccPath} parse`,
-            `${reportsFolder}`,
-            '-e json',
-            filePaths
-        ].join(' ');
+            'parse',
+            reportsFolder,
+            '-e', 'json',
+            ...filePaths
+        ];
     }
 
-    public getVersionCmdLine(): string | undefined {
-        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
-            ?? 'CodeChecker';
-
+    public getVersionCmdArgs(): string[] | undefined {
```

Comment:
When I started VSCode with your branch I got the following error:
![image](https://user-images.githubusercontent.com/6695818/154636432-8fab4852-37b1-454a-b534-4256ec47cbd3.png)

In the console I saw this:
```
>>> Process 'CodeChecker version' errored: spawn ~/CodeChecker/bin/CodeChecker ENOENT
>>> CodeChecker error while checking version
```

In the settings I set the CodeChecker binary path to ~/CodeChecker/bin/CodeChecker. Previously it worked so I think there is some bug in your patch.

Also please change this line to analyzer-version because the error message above is a little bit misleading (we do not run CodeChecker version command but CodeChecker analyzer-version):
https://github.com/Ericsson/CodecheckerVSCodePlugin/blob/7f628b85a014dbeaf7484a571baa5c6c92730360/src/backend/executor/process.ts#L17

Reply:
The problem is that my CodeChecker executable path started with `~` and the spawn doesn't work properly with such cases. See: https://github.com/nodejs/node/issues/684

Reply:
Updated the docs to say the setting requires an absolute path.

Reply:
We should still remove shell true.